Controller/NotifyHandler.php

```diff
@@ -245,4 +250,46 @@ public static function getNotificationUrl($oPaymentMethod)
             . 'index.php?cl=wcpg_notifyhandler&fnc=handleRequest&pmt='
             . PaymentMethod::getOxidFromSDKName($oPaymentMethod->getTransaction()->getConfigKey());
     }
+
+    /**
+     * Recursively tries to load the corresponding order
+     * Needed because oxid some times(i.e non 3d credit card payments) does not create the order
+     * before we get the success notification from the payment sdk
+     *
+     * @param Order  $oOrder
+     * @param string $sTransactionId
+     * @param int    $iIteration
+     *
+     * @return int   needed time in seconds to load order
+     *
+     * @since 1.3.0
+     */
+    private function _loadOrder(&$oOrder, $sTransactionId, $iIteration = 0)
```

Comment:
I think it would be cleaner if this method wouldn't depend on its own return value to make the polling work, e.g.:

```php
private function _loadOrder(&$oOrder, $sTransactionId, $iWaitTime = self::MAX_TIMEOUT_SECONDS)
{
    if ($iWaitTime > 0 && !$oOrder->loadWithTransactionId($sTransactionId)) {
        sleep(self::ITERATION_TIMEOUT_SECONDS);
        $this->_loadOrder($oOrder, $sTransactionId, $iWaitTime - self::ITERATION_TIMEOUT_SECONDS);

        return true;
    }

    return false;
}
```

Reply:
that code would not work
I cannot check if the timeout was reached, when I don't use the return value or another if statement

Reply:
You're passing it on to the next iteration in the last parameter. This should work just fine.

Reply:
no, because it's a recursive function